pkg/apiserver/registry/stats/nodelatencystats/rest_test.go

```diff
@@ -13,3 +13,200 @@
+				assert.Equal(t, tt.expectedObj, obj)
+			}
+		})
+	}
+}
+
+func TestRESTDelete(t *testing.T) {
+	tests := []struct {
+		name        string
+		summary     *statsv1alpha1.NodeLatencyStats
+		nodeName    string
+		expectedObj runtime.Object
+		expectedErr bool
+		err         error
+	}{
+		{
+			name: "delete summary",
+			summary: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+			nodeName: "node1",
+			expectedObj: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+			expectedErr: false,
+			err:         nil,
+		},
+		{
+			name: "delete summary not found",
+			summary: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+			nodeName:    "node2",
+			expectedObj: nil,
+			expectedErr: true,
+			err:         errors.NewNotFound(statsv1alpha1.Resource("nodelatencystats"), "node2"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := NewREST()
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			_, err := r.Create(ctx, tt.summary, nil, nil)
+			assert.NoError(t, err)
+			obj, deleted, err := r.Delete(ctx, tt.nodeName, nil, nil)
+			if tt.expectedErr {
+				assert.Equal(t, tt.err.(*errors.StatusError).ErrStatus.Code, err.(*errors.StatusError).ErrStatus.Code)
+			} else {
+				require.NoError(t, err)
+				assert.True(t, deleted)
+				assert.Equal(t, tt.expectedObj, obj)
+			}
+		})
+	}
+}
+
+func TestRESTList(t *testing.T) {
+	summary := &statsv1alpha1.NodeLatencyStats{
+		ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+		PeerNodeLatencyStats: nil,
+	}
+	options := &internalversion.ListOptions{
+		Limit:    10,
+		Continue: "",
+	}
+	expectedObj := &statsv1alpha1.NodeLatencyStatsList{
+		Items: []statsv1alpha1.NodeLatencyStats{
+			{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+		},
+	}
+
+	r := NewREST()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	_, err := r.Create(ctx, summary, nil, nil)
+	assert.Nil(t, err)
+	objs, err := r.List(ctx, options)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedObj, objs)
+}
+
+func TestRESTConvertToTable(t *testing.T) {
+	summary := &statsv1alpha1.NodeLatencyStats{
+		ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+		PeerNodeLatencyStats: nil,
+	}
+	expectedObj := &statsv1alpha1.NodeLatencyStats{
+		ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+		PeerNodeLatencyStats: nil,
+	}
+
+	r := NewREST()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	_, err := r.Create(ctx, summary, nil, nil)
+	assert.Nil(t, err)
+	obj, err := r.ConvertToTable(ctx, summary, nil)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedObj, obj.Rows[0].Object.Object)
```

Comment:
I don't think this test is doing anything
You need to populate `PeerNodeLatencyStats` with some data and validate the actual rows.

Reply:
Ok, I populated  `PeerNodeLatencyStats` in this case.